# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/protocol/http/nsAHttpTransaction.h
# Commit: 8e7c97a37073
# Full Hash: 8e7c97a37073a14ba832e2f5c27f2c11b7787ef5
# Author: Valentin Gosu <valentin.gosu@gmail.com>
# Date: 2023-03-15 16:26:49
# Regressor Bug: 1805372
# File Overlap Count: 1
# Description:
#   Bug 1805372 - Pass effectiveTRRMode to nsHttpChannel from socketTransport/DnsAndConnectSocket r=necko-reviewers,kershaw
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D164539
# ==============================================================================

diff -r 5abfd5ad2e35 -r 8e7c97a37073 netwerk/protocol/http/nsAHttpTransaction.h
--- a/netwerk/protocol/http/nsAHttpTransaction.h	Wed Mar 15 12:41:26 2023 +0000
+++ b/netwerk/protocol/http/nsAHttpTransaction.h	Wed Mar 15 13:00:46 2023 +0000
@@ -7,6 +7,8 @@
 
 #include "nsTArray.h"
 #include "nsWeakReference.h"
+#include "nsIRequest.h"
+#include "nsITRRSkipReason.h"
 
 #ifdef Status
 /* Xlib headers insist on this for some reason... Nuke it because
@@ -181,6 +183,8 @@
   virtual void ReuseConnectionOnRestartOK(bool) {}
   virtual void SetIsHttp2Websocket(bool) {}
   virtual bool IsHttp2Websocket() { return false; }
+  virtual void SetTRRInfo(nsIRequest::TRRMode aMode,
+                          TRRSkippedReason aSkipReason){};
 
   // We call this function if we want to use alt-svc host again on the next
   // restart. If this function is not called on the next restart the